main.sh

```diff
@@ -1,60 +1,122 @@
+	git checkout v0.8.1
+	make gvproxy
+	mkdir -p "$workspace/out"
+	mv ./bin/gvproxy "$workspace/out/gvproxy"
+	chmod +x "$workspace/out/gvproxy"
+	cd "$workspace"
+}
+
+_pack_output() {
+	if [[ -z $GOARCH ]] || [[ -z $GOOS ]]; then
+		echo "GOARCH and GOOS not set"
+		exit 100
+	fi
+
+	cd "$workspace/out"
+	tar -zcvf "$workspace/libexec-$GOOS-$GOARCH.tar.gz" .
+	cd "$workspace"
+}
+
+_get_vfkit() {
+	v_tag="v0.6.0"
+	wget https://github.com/crc-org/vfkit/releases/download/$v_tag/vfkit --output-document="$workspace/out/vfkit"
+	chmod +x "$workspace/out/vfkit"
+}
+
+_do_codesign() {
+	if [[ -z "$CODESIGN_IDENTITY" ]]; then
+		CODESIGN_IDENTITY="-"
+	fi
+
+	test -f "$workspace/out/gvproxy" && {
+		echo "Signing gvproxy..."
+		codesign --force --sign $CODESIGN_IDENTITY --options=runtime --timestamp $workspace/out/gvproxy
+	}
+
+	test -f "$workspace/out/vfkit" && {
+		echo "Signing krunkit..."
+		codesign --force --sign $CODESIGN_IDENTITY --options=runtime --timestamp --entitlements krunkit.entitlements $workspace/out/vfkit
+	}
+
+	find "$workspace/out" -name "*.dylib" -type f -exec sh -c "echo 'Set {} permission to 755'; chmod 755 {}" ';'
+	find "$workspace/out" -name "*.dylib" -type f -exec sh -c "echo 'Signing {}...'; codesign --force --sign $CODESIGN_IDENTITY --options=runtime --timestamp {}" ';'
+}
+
+_get_krunkit() {
+	cd "$workspace"
+	krun_url=https://github.com/containers/krunkit/releases/download/v0.1.4/krunkit-podman-unsigned-0.1.4.tgz
+	mkdir -p "$workspace/krunkit_tmp"
+	wget "$krun_url" --output-document="$workspace/krunkit_tmp/krunkit.tar.gz"
+	tar -xvf "$workspace/krunkit_tmp/krunkit.tar.gz" -C "$workspace/krunkit_tmp"
+	mv $workspace/krunkit_tmp/bin/* $workspace/krunkit_tmp/lib/* $workspace/out
+}
+
+build_darwin_arm64() {
+	rm -rf "$workspace/out"
+	mkdir -p "$workspace/out"
+	export GOOS=darwin
+	export GOARCH=arm64
+	echo "Build gvproxy"
+	_build_gvproxy
+	echo "Download krunkit"
+	_get_krunkit
+	echo "Do codesign"
+	_do_codesign
+	echo "Packup output"
+	_pack_output
+}
+
+build_darwin_amd64() {
+	# Clean out dir first
+	rm -rf "$workspace/out"
+	mkdir -p "$workspace/out"
+	export GOOS=darwin
+	export GOARCH=amd64
+	echo "Build gvproxy"
+	_build_gvproxy
+	echo "Download vfkit"
+	_get_vfkit
+	echo "Do codesign"
+	_do_codesign
+	echo "Packup output"
+	_pack_output
+}
+
+check_system_requires() {
+	go version >/dev/null || {
+		echo "Error: Go compiler not find, you need install go compile on your system"
+		exit 100
+	}
+}
+
+main() {
+	target_arch=$1
+	if [[ -z $target_arch ]]; then
+		echo "Error: missing target"
+		exit 100
+	fi
+
+	check_system_requires
+
+	workspace="$(pwd)"
+
+	if [[ $target_arch == arm64 ]] || [[ $target_arch == aarch64 ]]; then
+		echo "Building binaries for darwin arm64/aarch64"
+		build_darwin_arm64
+	elif [[ $target_arch == x86_64 ]] || [[ $target_arch == amd64 ]]; then
+		echo "Building binaries for darwin amd64/x86_64"
+		build_darwin_amd64
+	else
+		echo "Not support targer $target_arch"
+		exit 100
```

Comment:
```suggestion
		exit 2
```

![CleanShot 2024-12-26 at 10 34 25@2x](https://github.com/user-attachments/assets/3e05b24f-dd17-40c4-94eb-25ce89d8df9c)

We should try to align ourselves with mainstream software.

Reply:
I changed `exit 100` to `exit 2` but `exit 2` not just means `unrecognized option`, it can be 1 or any other number
```
➜  ~ gcc --error-params
gcc: error: unrecognized command-line option ‘--error-params’
gcc: fatal error: no input files
compilation terminated.
➜  ~ echo $?
1
``` 

A general convention reserves status values `128` and up for special purposes. In particular, the value `128` is used to indicate failure to execute another program in a subprocess but no one cares about that

Reply:
> I changed exit 100 to exit 2 but exit 2 not just means unrecognized option, it can be 1 or any other number

Yes, either 1 or 2 can be used here, but it's best not to use 100, as this would rather increase the cost of understanding.